extensions/packageManager/src/node/feeds/nuget/localNuGetFeed.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IExtensionRegistration } from '@bfc/extension-client';
+import { IFeed, IPackageQuery, IPackageDefinition } from '../feedInterfaces';
+import { INuGetPackage } from './nugetInterfaces';
+import { readdir, readFile } from 'fs-extra';
+import { parseStringPromise } from 'xml2js';
+import path from 'path';
+import * as semverSort from 'semver-sort';
+
+/**
+ * NuGet feed in local file system.
+ * @description Loads a NuGet package feed from the file system according to the NuGet documentation for local sources.
+ * Docs: https://docs.microsoft.com/en-us/nuget/hosting-packages/local-feeds.
+ */
+export class LocalNuGetFeed implements IFeed {
+  private language: string = 'c#';
+  private source: string = 'nuget';
+
+  /**
+   * Creates a local NuGet feed.
+   * @param composer Composer registration used for logging.
+   * @param url Url of the feed, represented usually by the file path to the NuGet file system feed.
+   */
+  constructor(private composer: IExtensionRegistration, private url: string) {}
+
+  /**
+   * Gets packages from local NuGet feed according to query request.
+   * @param query Query parameters.
+   * @todo Query parameters currently ignored. Implement query in file system feed if needed.
+   */
+  async getPackages(query?: IPackageQuery): Promise<IPackageDefinition[]> {
+    return await this.crawlLocalFeed(this.url);
+  }
+
+  /**
+   * Crawls a local feed starting from the root directory.
+   * @param url Root directory.
+   */
+  private async crawlLocalFeed(url: string): Promise<IPackageDefinition[]> {
+    // The local feed is expected to be in folders using the structure defined here:
+    // https://docs.microsoft.com/en-us/nuget/hosting-packages/local-feeds
+    // the line below will:
+    // * get a list of all the files at the specified url
+    // * extract only folders from that list
+    // * pass each one through the getPackageInfo function, which extracts metadata from the package
+    // * return a feed in the form that is used by nuget search API
+    const packages = await readdir(url, { withFileTypes: true });
+    const feedPromises: Promise<IPackageDefinition>[] = [];
+    for (const p of packages.filter((f) => f.isDirectory())) {
+      feedPromises.push(this.getPackageInfo(url, p.name));
+    }
```

Comment:
`const feedPromises = packages.filter((f) => f.isDirectory()).map((f) => this.packageInfo(url, f.name));` should work too.